activerecord/test/cases/relation/load_async_test.rb

```diff
@@ -5,7 +5,25 @@
 require "models/comment"
 
 module ActiveRecord
+  module WaitForAsyncTestHelper
+    private
+      def wait_for_async_query(relation, timeout: 5)
```

Comment:
Since the notification is no longer triggered in the background thread, we can no longer use a monitor to do synchronization. The best I could think of is a sleep loop that check the `FutureResult` state.